compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -313,9 +313,14 @@ object Splicer {
       case Call(fn, args) =>
         if (fn.symbol.isConstructor && fn.symbol.owner.owner.is(Package)) {
           interpretNew(fn.symbol, args.map(interpretTree))
+        } else if (fn.symbol.is(Module)) {
+          interpretModuleAccess(fn.symbol)
         } else if (fn.symbol.isStatic) {
-          if (fn.symbol.is(Module)) interpretModuleAccess(fn.symbol)
-          else interpretStaticMethodCall(fn.symbol, args.map(arg => interpretTree(arg)))
+          val module = fn.symbol.owner
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
+        } else if (fn.qualifier.symbol.is(Module) && fn.qualifier.symbol.isStatic) {
+          val module = fn.qualifier.symbol.moduleClass
+          interpretStaticMethodCall(module, fn.symbol, args.map(arg => interpretTree(arg)))
```

Comment:
Accessing `fn.qulifier` is dangerous (e.g., with import `Ident.qualifier.symbol` will return `NoSymbol`), it's more reliable to check types of `fun`, only handle the `TermRef` case. Then it seems the last two `else if` can be refactored and merged.

Reply:
Here we never traverse imports, the only refTree are the ones in expressions.

Reply:
Sorry, I was not clear, I mean the case like the following:

```Scala
import o._
f(1)

// in constrast to
o.f(1)
```

The code seems to only handle the 2nd case, not the 1st one.

Reply:
Oh, the second one will be generalized to the evaluation of any prefix in the future.